src/actions/expressions.js

```diff
@@ -38,6 +37,7 @@ export function addExpression(input: string) {
       return dispatch(evaluateExpression(expression));
     }
 
+    const expressionError = await parser.hasSyntaxError(input);
     dispatch({ type: "ADD_EXPRESSION", input, expressionError });
```

Comment:
Sorry accidentally deleted both yours and one I added accidentally @jasonLaster. So the reducer prevents the creation of the expression. So no expression created.